tests/decide_gi.py

```diff
@@ -1,41 +1,29 @@
-from input_output.file_output import load_graph_list
+import time
+
+from input_output.file_output import load_graph_list, create_csv_file, write_csv_line
 from input_output.sys_output import fail
+from input_output.sys_output import passed
 from supporting_components.graph_io import *
 from algorithms.decide_gi import is_balanced_or_bijected
 from algorithms.color_initialization import degree_color_initialization
-from algorithms.color_refinement import color_refinement
-
-# Expected test results
-total_tests = 22
-
-files = ['colorref_smallexample_2_49', 'colorref_smallexample_4_7', 'colorref_smallexample_6_15']
-
-# Dictionary containing expected outcomes.
-# Sourced from canvas.utwente.nl
-# Structure:
-# The dictionary is keyed on filename (str)
-# This dictionary contains for each key:
-#   A dictionary keyed on graph no. (int)
-#   This dictionary contains:
-#       A dictionary container on graph no. (int)
-#       This dictionary contains:
-#           Tuple containing : (is_balanced, is_bijected) ((bool, bool))
-
-expected_results = dict()
-expected_results['colorref_smallexample_2_49'] = {0: {1: (True, True)}}
-expected_results['colorref_smallexample_4_7'] = {0:  {1: (False, False), 2: (True, False), 3: (False, False)},
-                                                 1:  {2: (False, False), 3: (True, True)},
-                                                 2:  {3: (False, False)}
-                                                 }
-expected_results['colorref_smallexample_6_15'] = {0: {1: (True, True), 2: (False, False), 3: (False, False), 4: (False, False), 5: (False, False)},
-                                                  1: {2: (False, False), 3: (False, False), 4: (False, False), 5: (False, False)},
-                                                  2: {3: (True, True), 4: (False, False), 5: (False, False)},
-                                                  3: {4: (False, False), 5: (False, False)},
-                                                  4: {5: (True, False)}
-                                                  }
-
-
-def test_balanced_or_bijected(graph: 'Graph', is_balanced: 'Bool', is_bijected: 'Bool'):
+import fun_provider
+
+class Decide_gi:
+
+    def unittest(self):
+        # Swap in the color refinement object Default_color_refinement().
+        # This object has .color_refine(), which is what this function expects for color refinement during test.
+        return _test_decide_gi(test_name='decide_gi_unit', color_refine_object = fun_provider.Default_color_refinement())
+
+    # Create a function for the object to be called externally
```

Comment:
Ik dacht dat we bij de unittests gewoon voor 1 van de color_refinements gingen kiezen?
Want nu zijn beide opties er toch, terwijl ik ook het idee heb dat deze methoden nergens gebruikt worden? Dus deze zijn nu overbodig?

Reply:
Deze methoden gebruik ik niet op dit moment, ze staan er alleen als voorbeeld.